service/src/test/java/org/mskcc/smile/service/SampleServiceTest.java

```diff
@@ -290,57 +290,46 @@ public void testPatientSamplesCountMatchExpectedValues() throws Exception {
-    //        sampleService.updateSampleMetadata(updatedMetadata);
-    //        List<SampleMetadata> sampleMetadataHistoryAfterUpdate = sampleService
-    //                .getResearchSampleMetadataHistoryByIgoId(igoId);
-    //        Assertions.assertThat(sampleMetadataHistoryAfterUpdate.size()).isEqualTo(2);
-    //
-    //        // confirm that new sample metadata was persisted and that there is an older sample
-    //        // metadata with the date '2000-06-10' that we basically inserted into the
-    //        // history for this sample
-    //        Boolean hasMockOldMetadata = Boolean.FALSE;
-    //        for (SampleMetadata sm : sampleMetadataHistoryAfterUpdate) {
-    //            if (sm.getImportDate().equals("2000-06-10")) {
-    //                hasMockOldMetadata = Boolean.TRUE;
-    //                break;
-    //            }
-    //        }
-    //        Assertions.assertThat(hasMockOldMetadata).isTrue();
-    //
-    //        // confirms that both methods return the same latest metadata and
-    //        // same cmo sample label corresponding to it
-    //        // the most up-to-date cmo label is C-MP789JR-N001-d based on mocked test data
-    //        SmileSample updatedSample = sampleService.getResearchSampleByRequestAndIgoId(requestId, igoId);
-    //        Assertions.assertThat(updatedSample.getLatestSampleMetadata().getCmoSampleName())
-    //                .isEqualTo("C-MP789JR-N001-d");
-    //        SampleMetadata latestMetadata =
-    //                sampleRepository.findLatestSampleMetadataBySmileId(updatedSample.getSmileSampleId());
-    //        Assertions.assertThat(latestMetadata.getCmoSampleName()).isEqualTo("C-MP789JR-N001-d");
-    //    }
+    /**
+     * Test that the latest metadata returns the expected cmo label,
+     * even when metadata is added to a sample that a fake date that precedes the "latest" date
+     * @throws Exception
+     */
+    public void testFindLatestSampleMetadataAfterSavingNewPredatedSample() throws Exception {
+        String requestId = "MOCKREQUEST1_B";
+        String igoId = "MOCKREQUEST1_B_2";
+        String importDate = "2000-06-10";
+
+        SampleMetadata sampleMetadata = sampleService.getResearchSampleByRequestAndIgoId(
+                requestId, igoId).getLatestSampleMetadata();
+
+        SmileSample predatedSample = new SmileSample();
+        SampleMetadata predatedSampleMetadata = new SampleMetadata();
+        predatedSampleMetadata.setPrimaryId(igoId);
+        predatedSampleMetadata.setIgoRequestId(requestId);
+        predatedSampleMetadata.setCmoPatientId(sampleMetadata.getCmoPatientId());
+        predatedSampleMetadata.setImportDate(importDate);
+        predatedSampleMetadata.setCmoSampleName("C-OLDSAMPLELABEL-T11");
+        predatedSample.addSampleMetadata(predatedSampleMetadata);
+
+        sampleService.updateSampleMetadata(predatedSampleMetadata);
```

Comment:
with this and [this](https://github.com/mskcc/smile-server/pull/719/files#diff-6827768a6c6e39015875960a323fe05ff9844a0c428fd7aa8f0e133ba70cb148R323) isn't the sample data being saved twice?

Reply:
Hmm it wouldn't even get saved the second time. Should i make then 2 separate tests?

Reply:
I mean it wouldn't get saved the second time because there wouldnt be any new changes to persist but we should still remove the extra call to avoid confusion on what should be going on